[voltdb] Switch to native VoltDB Python client (#24714)
DataDog/integrations-core: voltdb/README.md

2. Edit the `voltdb.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your VoltDB performance data. See the [sample voltdb.d/conf.yaml][4] for all available configuration options.

>     The integration supports two transports. Pick the one that matches your network topology:

Reviewer: ```suggestion
    The integration supports two transports. Use the one that matches your network topology:
```

>     **Native binary client** - direct connection to a database node on the VoltDB client port (default `21212`), using the [VoltDB Python client][12]. Recommended when the Agent host can reach the database nodes directly:

Reviewer: ```suggestion
    **Native binary client** - direct connection to a database node on the VoltDB client port (default `21212`), using the [VoltDB Python client][12]. Datadog recommends the native binary client transport when the Agent host can reach the database nodes directly:
```

    ```yaml
    init_config:

    instances:
      - host: localhost
        port: 21212
        username: datadog-agent
        password: "<PASSWORD>"
    ```

    For failover across cluster members, use `hosts` instead of `host`. The Agent connects to the first reachable entry and silently fails over to the others if the active node becomes unavailable:

    ```yaml
    instances:
      - hosts:
          - voltdb-1.example:21212
          - voltdb-2.example:21212
          - voltdb-3.example:21212
        username: datadog-agent
        password: "<PASSWORD>"
    ```

>     **HTTP/JSON via the VoltDB Management Center (VMC)** - useful when database nodes aren't directly reachable but the VMC endpoint is, or when you prefer to keep the existing HTTP/JSON wire format. Set `url` to the VMC endpoint:

Reviewer: ```suggestion
    **HTTP/JSON using the VoltDB Management Center (VMC)** - useful when database nodes aren't directly reachable but the VMC endpoint is, or when you prefer to keep the existing HTTP/JSON wire format. Set `url` to the VMC endpoint:
```

    ```yaml
    instances:
      - url: http://vmc.example.com:8080
        username: datadog-agent
        password: "<PASSWORD>"
    ```

>     The check picks the transport based on which option is set: `url` selects HTTP mode, `host`/`hosts` selects native mode. HTTP mode keeps all the options earlier releases supported (`password_hashed`, `tls_cert`, `tls_ca_cert`, `tls_verify`, `proxy`, `headers`, etc.) - see the [sample config][4] for the full list.

Reviewer: ```suggestion
    The check picks the transport based on which option is set: `url` selects HTTP mode, `host`/`hosts` selects native mode. HTTP mode keeps all options supported by earlier releases  (`password_hashed`, `tls_cert`, `tls_ca_cert`, `tls_verify`, `proxy`, `headers`, etc.). See the [sample config][4] for the full list.
```

>     **Backwards compatibility**: existing configurations that point `url` at the legacy HTTP endpoint continue to work without changes. This release adds the native binary transport as an option; it does not remove the HTTP one. The `url`-style config still emits the same metrics and service checks against the same VMC or HTTP-enabled database node it always pointed at.

Reviewer: ```suggestion
    **Backwards compatibility**: existing configurations that point `url` at the legacy HTTP endpoint continue to work without changes. This release adds the native binary transport as an option; it does not remove the HTTP one. The `url`-style config still emits the same metrics and service checks against the same VMC- or HTTP-enabled database node it always pointed to.
```

3. [Restart the Agent][5].

#### TLS support

> If [TLS/SSL][6] is enabled on the VoltDB client port, set `use_ssl: true` and point `ssl_config_file` at a VoltDB SSL properties file that describes how to locate the truststore (and optionally a client keystore for mutual TLS):

Reviewer: ```suggestion
If [TLS/SSL][6] is enabled on the VoltDB client port, set `use_ssl: true` and point `ssl_config_file` to a VoltDB SSL properties file that describes how to locate the truststore (and optionally a client keystore for mutual TLS):
```

```yaml
instances:
  - host: localhost
    port: 21212
    username: datadog-agent
    password: "<PASSWORD>"
    use_ssl: true
    ssl_config_file: /etc/voltdb/ssl.properties
```

The properties file is the same format VoltDB's own tools (`sqlcmd`, `voltadmin`) consume. The native Python client supports Java keystores (`.jks`), PKCS12 (`.p12`/`.pfx`), and PEM. A minimal one-way TLS configuration looks like:

```properties
# /etc/voltdb/ssl.properties
truststore=/etc/voltdb/certs/truststore.jks
truststorepassword=<TRUSTSTORE_PASSWORD>
```

For mutual TLS, also add a keystore that identifies the Agent to the server:

```properties
truststore=/etc/voltdb/certs/truststore.jks
truststorepassword=<TRUSTSTORE_PASSWORD>
keystore=/etc/voltdb/certs/agent-keystore.jks
keystorepassword=<KEYSTORE_PASSWORD>
```

> If you have a PEM CA bundle instead of a Java keystore, you can either point `ssl_config_file` directly at the PEM file (it is treated as the truststore), or reference it explicitly with `cacerts=<PATH>` inside the properties file.

Reviewer: ```suggestion
If you have a PEM CA bundle instead of a Java keystore, you can either point `ssl_config_file` directly at the PEM file, which is treated as the truststore, or reference it explicitly with `cacerts=<PATH>` inside the properties file.
```
